### ISSUE TITLE
Failed to spawn: unable to start helper

### ISSUE THREAD

**Issue by Xeon8297** [frida/frida-core#381]
I have an error "Failed to spawn: unable to start helper"
command im using: frida -U Gadget -l script.js --no-pause --runtime=v8
this error appeared since frida v15
Anyone knows how to fix that?

**Comment by brawler888** [comment]
I know! need a secret technology! "You are SuperSegaMega Dudnik"

**Comment by svengong** [comment]
the same to u.
```
frida-ps -U
Failed to enumerate processes: unable to start helper
```

**Comment by mxrch** [comment]
python3 -m pip uninstall frida frida-tools
python3 -m pip install frida==14.2.18 frida-tools=9.2.5

:)

**Comment by oleavr** [comment]
Please include Android OS version and device type. This error means it's failing to connect to a frida-server on the Android side, and is falling back to non-rooted mode by launching a .dex – and this is what's failing.

**Comment by oleavr** [comment]
So if you're seeing this you probably forgot to upgrade the device side Frida to v15. But regardless I'd like to get the non-rooted codepath working better too, so OS version details would be helpful in reproducing this.

**Comment by mxrch** [comment]
Not very cool to delete my temporary workaround

**Comment by oleavr** [comment]
@mxrch Apologies, and I agree it's uncool, but OTOH it was misleading others to run old buggy versions that have worse bugs already fixed in 15.x – I'm already spread too thin and don't want to also have to deal with supporting old releases. 😅

This new error in v15 is confusing and will be fixed soon, but it simply means that the frida-server / frida-gadget on the remote side is incompatible (likely 14.x instead of 15.x). The non-rooted Android support was also highly experimental in 14.x, but nobody noticed it because the remote Frida was protocol-compatible.

**Comment by oleavr** [comment]
Sorry for being so heavy-handed here, I will do better in the future. ❤️ This is really all my fault for not setting up proper issue templates, and also not taking the time to implement special error-handling for new-to-old scenarios. (Which I really should have done given that I've changed the protocol so radically with Frida 15.)

**Comment by oleavr** [comment]
Fixed in 15.0.9 – in the sense that we're now fault-tolerant and start ignoring the non-rooted code-path if we fail to start the helper. Since I haven't heard back from anyone able to reproduce this, I will close this for now – hopefully someday somebody able to reproduce this can help fix it. Until then, things are at least in overall better shape than they were in Frida 14, which is what matters – that we suck a little bit less with every new release 😄
